### PR TITLE
Re-align SetTargetPopulation priorities

### LINKED ISSUE #2153: Population effect priorities are broken
https://github.com/freeorion/freeorion/commit/9f02a55212b3f6f7193228c5fd91bc116d9102f9 by @geoffthemedio changed the order of growth effects. At the very least, species modifier priorities are changed.

There is a comment by @Dilvish-fo pointing that out but I can't see the issue resolved...(https://github.com/freeorion/freeorion/commit/9f02a55212b3f6f7193228c5fd91bc116d9102f9#r27525956)

Even if the changes are intended for whatever reason, this means that at the very least, the AI calculations are currently significantly broken and must be updated.

### PR DESCRIPTION
**Pull request by Dilvish-fo**
-Per discussion of #2153

Goes with the 4 tier proposal from Morlic, although I did not rename the existing priorities, I simply added one new one.

-Fixes #2153

### CODE REVIEW

**Comment by Dilvish-fo** [comment]
Those python lint errors are unrelated to the present PR, and are fixed in #2164 so I'd suggest we ignore them here.

File: default/scripting/specials/planet/TEMPORAL_ANOMOLY.focs.txt [hunk 1]
```diff
@@ -15,12 +15,12 @@ Special
                 Source
                 Focus type = "FOCUS_RESEARCH"
             ]
-            priority = [[EARLY_PRIORITY]]
+            priority = [[EARLY_TARGET_POPULATION_POST_SCALING_PRIORITY]]
```
Comment (Morlic-fo) [inline default/scripting/specials/planet/TEMPORAL_ANOMOLY.focs.txt:18]: Looks wrong.

File: default/scripting/species/common/population.macros [hunk 2]
```diff
@@ -173,7 +173,7 @@ PHOTOTROPHIC_BONUS
                 TargetPopulation low = 0
             ]
             accountinglabel = "VERY_BRIGHT_STAR"
-            priority = [[EARLY_POPULATION_PRIORITY]]
+            priority = [[EARLY_TARGET_POPULATION_POST_SCALING_PRIORITY]]
```
Comment (Morlic-fo) [inline default/scripting/species/common/population.macros:176]: As this depends on the TargetPopulation, should probably be given a later priority?

File: default/scripting/species/common/population.macros [hunk 3]
```diff
@@ -185,7 +185,7 @@ PHOTOTROPHIC_BONUS
                 TargetPopulation low = 0
             ]
             accountinglabel = "BRIGHT_STAR"
-            priority = [[EARLY_POPULATION_PRIORITY]]
+            priority = [[EARLY_TARGET_POPULATION_POST_SCALING_PRIORITY]]
```
Comment (Morlic-fo) [inline default/scripting/species/common/population.macros:188]: same here

File: default/scripting/species/common/xenophobic.macros [hunk 4]
```diff
@@ -53,7 +53,7 @@ XENOPHOBIC_SELF
             ]
             stackinggroup = "XENOPHOBIC_POP_SELF"
             accountinglabel = "XENOPHOBIC_LABEL_SELF"
-            priority = [[LATE_PRIORITY]]
+            priority = [[EARLY_TARGET_POPULATION_POST_SCALING_PRIORITY]]
```
Comment (Morlic-fo) [inline default/scripting/species/common/xenophobic.macros:56]: As effect depends on target population, should probably be given a later priority?

**Review by Morlic-fo — CHANGES_REQUESTED** [review]
Missing stuff:
* Orbital habitation to POST
* Planetary Ecology to POST

On more detailed review of the matter, should probably add another layer for effects that depend on target pop value.

File: default/scripting/species/common/population.macros (hunk 2 above)
Comment (Dilvish-fo) [reply default/scripting/species/common/population.macros:176]: As an additive adjustment post scaling, it doesn't need to be made later than other post-scaling additions, but as I noted in the Issue, your handling of this is fine as well.

File: default/scripting/specials/planet/TEMPORAL_ANOMOLY.focs.txt (hunk 1 above)
Comment (Dilvish-fo) [reply default/scripting/specials/planet/TEMPORAL_ANOMOLY.focs.txt:18]: Yes.  I don't think this difference actually matters, but I did note it in the Issue thread, it was a mistaken change that should not remain.

File: default/scripting/species/common/xenophobic.macros (hunk 4 above)
Comment (Dilvish-fo) [reply default/scripting/species/common/xenophobic.macros:56]: in particular, as the effect depends on the running value of the target population, not the initial value of TargetPopulation, yes, it is best handled as a later priority as you did.

**Comment by Dilvish-fo** [comment]
Just to note here, as I indicated in the Issue discussion, I'll be closing this as superseded as soon as Morlic posts his PR.

**Pull request closed without merging**